### PR TITLE
fix: Adapt async example to non-polymoprhic compare

### PR DESCRIPTION
**Pull request by psafont**
Also replaces travis with github actions for testing.

It's annoying how the checks won't appear here: https://github.com/psafont/xen-api-client/runs/1689674258

### CODE REVIEW

**Review by benjamreis — APPROVED** [review]

**Review by lippirk — APPROVED** [review]
I don't quite follow why `events = []`, or why we need to use the base version of `=`, but the behaviour ought to be the same

**Comment by psafont** [comment]
> I don't quite follow why the normal `=` from stdlib is not sufficient, but the behaviour ought to be the same

It's not from Stdlib, the Core is opened at the top. Since 0.13.0 the equality is non-polymorphic and by default only compares integers.

The package failing doesn't get built for xenserver. This means that when the janestreet packages were updated we missed this issue.

File: async_examples/event_test.ml [hunk 1]
```diff
@@ -76,7 +76,7 @@ let watch_events rpc session_id =
     let open Event_types in
     Event.from ~rpc ~session_id ~classes:["*"] ~token:"" ~timeout:0. >>= fun rpc ->
     let e = event_from_of_rpc rpc in
-    if e.events = [] then error "Empty list of events";
+    if List.length e.events = 0 then error "Empty list of events";
```
Comment (lindig) [inline async_examples/event_test.ml:79]: Why is this needed or better?

**Review by lindig — APPROVED** [review]

File: async_examples/event_test.ml (hunk 1 above)
Comment (psafont) [reply async_examples/event_test.ml:79]: `=` is an alias to the monomorphic `Core.Int.(=)` so it cannot be used here to compare lists.

Alternatives are:
1. Use a prefixed equal (something like `List.equal Event.equal e.events []`), but I don't think Event.equal is of easy access.
2. Remove the `Open.Core`. This breaks quite a few lines in the file so I avoided it.
3. Use length to be able to use integer comparison as is. I chose this because it seemed like the less disruptive choice.
